spring-data-relational/src/main/java/org/springframework/data/relational/repository/query/RelationalExampleMapper.java

```diff
@@ -64,92 +72,194 @@ public <T> Query getMappedExample(Example<T> example) {
+		// Criteria, assemble!
+		Criteria criteria = Criteria.empty();
 
-			if (property.isCollectionLike() || property.isMap()) {
-				return;
-			}
+		for (Criteria propertyCriteria : criteriaBasedOnProperties) {
 
-			if (matcherAccessor.isIgnoredPath(property.getName())) {
-				return;
+			if (example.getMatcher().isAllMatching()) {
+				criteria = criteria.and(propertyCriteria);
+			} else {
+				criteria = criteria.or(propertyCriteria);
 			}
+		}
+
+		return Query.query(criteria);
+	}
+
+	private <T> @NotNull List<Criteria> buildCriteriaRecursive( //
+			RelationalPersistentEntity<?> persistentEntity, //
+			ExampleMatcherAccessor matcherAccessor, //
+			PersistentPropertyAccessor<T> probePropertyAccessor //
+	) {
+		final List<Criteria> criteriaBasedOnProperties = new ArrayList<>();
+
+		persistentEntity.doWithProperties((PropertyHandler<RelationalPersistentProperty>) property -> {
+			potentiallyEnrichCriteriaByProcessingProperty(
+					null,
+					matcherAccessor,  //
+					probePropertyAccessor, //
+					property, //
+					criteriaBasedOnProperties //
+			);
+		});
+		return criteriaBasedOnProperties;
+	}
+
+	/**
+	 * Analyzes the incoming {@code property} and potentially enriches the {@code criteriaBasedOnProperties} with the new
+	 * {@link Criteria} for this property.
+	 * <p>
+	 * This algorithm is recursive in order to take the embedded properties into account. The caller can expect that the result
+	 * of this method call is fully processed subtree of an aggreagte where the passed {@code property} serves as the root.
+	 *
+	 * @param propertyPath the {@link PropertyPath} of the passed {@code property}.
+	 * @param matcherAccessor the accessor for the original {@link ExampleMatcher}.
+	 * @param entityPropertiesAccessor the accessor for the properties of the current entity that holds the given {@code property}
+	 * @param property the property under analysis
+	 * @param criteriaBasedOnProperties the {@link List} of criteria objects that potentially gets enriched as a
+	 *                                  result of the incoming {@code property} processing
+	 */
+	private <T> void potentiallyEnrichCriteriaByProcessingProperty(
+			@Nullable PropertyPath propertyPath,
+			ExampleMatcherAccessor matcherAccessor, //
+			PersistentPropertyAccessor<T> entityPropertiesAccessor, //
+			RelationalPersistentProperty property, //
+			List<Criteria> criteriaBasedOnProperties //
+	) {
+
+		// QBE do not support queries on Child aggregates yet
+		if (property.isCollectionLike() || property.isMap()) {
+			return;
+		}
+
+		PropertyPath currentPropertyPath = resolveCurrentPropertyPath(propertyPath, property);
+		String currentPropertyDotPath = currentPropertyPath.toDotPath();
+
+		if (matcherAccessor.isIgnoredPath(currentPropertyDotPath)) {
+			return;
+		}
 
+		Object actualPropertyValue = entityPropertiesAccessor.getProperty(property);
+
+		if (property.isEmbedded() && actualPropertyValue != null) {
```

Comment:
Ideally, we don't distinguish based on whether the property is an embedded one but rather whether it is an entity. At some point it should be possible to express criteria based across objects that are part of the aggregate (like it is done in JPA, we consider embeddables and references).